cmd/loki-canary-test/canary/client_test.go

```diff
@@ -0,0 +1,288 @@
+package canary_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/grafana/loki/cmd/loki-canary-test/canary"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/assert"
+)
+
+const sumEntriesQuery = "sum(loki_canary_entries_total)"
+const sumEntriesMissingQuery = "sum(loki_canary_missing_entries_total)"
+
+func TestCanary(t *testing.T) {
```

Comment:
This is not run as part of the main is it?

Reply:
switch to not having main and running everything as a `go test`